internal/controller/hooks/hooks_util.go

```diff
@@ -186,3 +255,51 @@ func getOpHookTimeoutValue(hook *kubeobjects.HookSpec) int {
 	// 300s is the default value for timeout
 	return defaultTimeoutValue
 }
+
+func getMatchingUnstructedObjs(uList *unstructured.UnstructuredList, re *regexp.Regexp) []client.Object {
+	objs := make([]client.Object, 0)
+
+	for _, uObj := range uList.Items {
+		if re.MatchString(uObj.GetName()) {
+			objs = append(objs, &uObj)
+		}
```

Comment:
Same for these four functions as above. A helper function with common code is cleaner.